homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,54 @@
+#include "compression.hpp"
+
+#include <algorithm>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedVector;
+    compressedVector.reserve(width * height);
+    auto nextCharacter = ' ';
+
+    std::for_each(bitmap.cbegin(), bitmap.cend(), [&](const auto& line) {
+        std::for_each(line.cbegin(), line.cend(), [&](const auto character) {
```

Comment:
for_each for me is similar to for. You can use transform :) or std::generate (you will capture iterator and then jump between different numbers and return proper pair :) )